qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -1237,6 +1243,133 @@ def _inject_site_specific_quirks(self):
+                lambda: message.info("Please touch your device now."),
+            QWebEngineWebAuthUxRequest.WebAuthUxState.SelectAccount:
+                self._ux_account_selection,
+            QWebEngineWebAuthUxRequest.WebAuthUxState.Completed:
+                self._ux_request_completed,
+            QWebEngineWebAuthUxRequest.WebAuthUxState.Cancelled:
+                self._ux_request_cancelled,
+            QWebEngineWebAuthUxRequest.WebAuthUxState.RequestFailed:
+                self._ux_request_failed
+        }[state]()
+
+    def _ux_pin_request(self):
+        assert self._request is not None
+        log.webview.debug("Collect Webuth pin for "
+                          f"{self._request.relyingPartyId()}")
+        answer = self._verification_required(self._request.relyingPartyId())
+        if answer is not None:
+            log.webview.debug("User verification accepted by user")
+            self._request.setPin(answer)
+        else:
+            log.webview.debug("User verification aborted by user")
+            self._request.cancel()
+
+    def _ux_account_selection(self):
+        assert self._request is not None
+        log.webview.debug("Select Webuth account for "
+                          f"{self._request.relyingPartyId()}")
+        answer = self._select_account(self._request.relyingPartyId(),
+                                      self._request.userNames())
+        if answer is not None:
+            log.webview.debug(f"Username selection accepted by user: {answer}")
+            self._request.setSelectedAccount(answer)
+        else:
+            log.webview.debug("Username selection aborted by user")
+            self._request.cancel()
+
+    def _ux_request_completed(self):
+        message.info("User verification completed.")
+        self._request = None
+
+    def _ux_request_cancelled(self):
+        message.info("User verification cancelled.")
+        self._tab.abort_questions.emit()
+        self._request = None
+
+    def _ux_request_failed(self):
+        assert self._request is not None
+        log.webview.debug("Webauth request failed for "
+                          f"{self._request.relyingPartyId()}: "
+                          f"{self._request.requestFailureReason()}")
+
+        reason = {
+            QWebEngineWebAuthUxRequest.RequestFailureReason.Timeout:
+                "The request timeout out.",
+            QWebEngineWebAuthUxRequest.RequestFailureReason.KeyNotRegistered:
+                "The device is not registered.",
+            QWebEngineWebAuthUxRequest.RequestFailureReason.KeyAlreadyRegistered:
+                "You already registered this device.",
+            QWebEngineWebAuthUxRequest.RequestFailureReason.SoftPinBlock:
+                "The device is locked because the wrong PIN was entered too many times.",
+            QWebEngineWebAuthUxRequest.RequestFailureReason.HardPinBlock:
+                "The device is locked because the wrong PIN was entered too many times.",
```

Comment:
The Qt docs say (emphasis mine):

> The authenticator is blocked as the user entered the wrong key many times **and reset the PIN to use the specific authenticator again.**

should we somehow differentiate soft/hard block via the error message we show to the user?